polaris-react/src/components/DataTable/components/Cell/Cell.tsx

```diff
@@ -156,31 +163,35 @@ export function Cell({
   const headingMarkup = header ? (
     <th
       {...headerCell.props}
+      aria-sort={sortDirection}
       {...colSpanProp}
       ref={setRef}
       className={className}
       scope="col"
-      aria-sort={sortDirection}
-      style={nthColumn ? {minWidth: firstColumnMinWidth} : {}}
+      style={{...minWidthStyles}}
     >
       {columnHeadingContent}
     </th>
   ) : (
     <th
-      style={{minWidth: firstColumnMinWidth}}
+      {...colSpanProp}
+      ref={setRef}
       className={className}
       scope="row"
-      {...colSpanProp}
-      ref={(ref) => {
-        setRef(ref);
-      }}
+      style={{...minWidthStyles}}
     >
-      <TruncatedText className={styles.TooltipContent}>{content}</TruncatedText>
+      {truncate ? (
+        <TruncatedText className={styles.TooltipContent}>
+          {content}
+        </TruncatedText>
+      ) : (
+        content
+      )}
```

Comment:
Now we only truncate if we actually have the `truncate` prop. @brianshen1990, this is what fixes the issue you were encountering.